### PR TITLE
[#2119] Update `VaDataTable` sorting with `sortingOptions` prop

### LINKED ISSUE #2119: how to allow only sort data table by ASC OR DESC not to null?
**Vuestic-ui version:** 1.4.6

**Steps to reproduce**

**What is the expected behavior?**

**What is the current behavior?**

**Other information**

### PR DESCRIPTION
**Pull request by RVitaly1978**
## Description
close #2119 

changes:
- [x] add `sortingOptions` prop that defines order of sorting
- [x] update doc example, locales

chrome-capture-2022-6-28
 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/src/page-configs/ui-elements/data-table/examples/Sorting.vue [hunk 1]
```diff
@@ -168,7 +168,7 @@ export default defineComponent({
     ]
 
     const columns = [
-      { key: 'id', sortable: true },
+      { key: 'id', sortable: true, sortingOptions: ['desc', 'asc'] },
```
Comment (rustem-nasyrov) [inline packages/docs/src/page-configs/ui-elements/data-table/examples/Sorting.vue:171]: Don't you need to also add `null`?

Comment (RVitaly1978) [reply packages/docs/src/page-configs/ui-elements/data-table/examples/Sorting.vue:171]: For example, we don't use `null` here as mentioned in issue #2119

File: packages/ui/src/components/va-data-table/hooks/useColumns.ts [hunk 2]
```diff
@@ -2,17 +2,42 @@ import { computed } from 'vue'
...
   columns: DataTableColumnSource[]
   items: DataTableItem[]
+  sortingOptions: DataTableSortingOptions
   [prop: string]: unknown
 }
 
-export const buildTableColumn = (source: DataTableColumnSource, initialIndex: number): DataTableColumnInternal => {
+export const sortingOptionsValidator = (options: DataTableSortingOptions) => (
+  (options.length === 2 || options.length === 3) &&
+  (options.length === [...new Set(options)].length) &&
```
Comment (aluarius) [inline packages/ui/src/components/va-data-table/hooks/useColumns.ts:22]:
```suggestion
  (options.length === new Set(options).size) &&
```

**Review by aluarius — APPROVED** [review]

File: packages/ui/src/components/va-data-table/hooks/useColumns.ts (hunk 2 above)
Comment (m0ksem) [reply packages/ui/src/components/va-data-table/hooks/useColumns.ts]: Woah

File: packages/docs/src/locales/en/en.json [hunk 3]
```diff
@@ -900,6 +900,7 @@
         "filterMethod": "A custom filtering function. The function takes the initial value of the currently checked cell (the `source` is a formal parameter) and should return a `boolean` value, indicating whether to include the row containing this cell or not",
         "sortBy": "Sets the column to sort by. Works through the `v-model`",
         "sortingOrder": "Sets the sorting order. Works through the `v-model`",
+        "sortingOptions": "An array from a set of values (`'asc'`, `'desc'`, `null`) that specifies the order and allowed sort directions.",
```
Comment (dogfrogfog) [inline packages/docs/src/locales/en/en.json:903]: extra `.` in the end ?

File: packages/ui/src/components/va-data-table/hooks/useSortable.ts [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import { computed, ref, Ref, watch } from 'vue'
 
-import { DataTableColumnInternal, DataTableRow, DataTableItem, DataTableSortingOrder } from '../types'
+import {
```
Comment (dogfrogfog) [inline packages/ui/src/components/va-data-table/hooks/useSortable.ts:3]: you can try to take adventage of [that](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html)

File: packages/ui/src/components/va-data-table/hooks/useColumns.ts [hunk 5]
```diff
@@ -2,17 +2,42 @@ import { computed } from 'vue'
...
   items: DataTableItem[]
+  sortingOptions: DataTableSortingOptions
   [prop: string]: unknown
 }
 
-export const buildTableColumn = (source: DataTableColumnSource, initialIndex: number): DataTableColumnInternal => {
+export const sortingOptionsValidator = (options: DataTableSortingOptions) => (
+  (options.length === 2 || options.length === 3) &&
+  (options.length === new Set(options).size) &&
+  (options.every((option) => ['asc', 'desc', null].includes(option)))
```
Comment (dogfrogfog) [inline packages/ui/src/components/va-data-table/hooks/useColumns.ts:23]: hard to get it right

File: packages/docs/src/locales/en/en.json (hunk 3 above)
Comment (RVitaly1978) [reply packages/docs/src/locales/en/en.json:903]: After discussion, we decided to use `.` at the end of sentences in tables.

**Pull request merged**